sdk/storage/Azure.Storage.Blobs/src/Models/BlobBlock.cs

```diff
@@ -12,6 +13,28 @@ namespace Azure.Storage.Blobs.Models
     [CodeGenModel("Block")]
     public readonly partial struct BlobBlock : IEquatable<BlobBlock>
     {
+        // All properties are rebuilt by hand in this class to avoid compile error CS0282
+
+        /// <summary>
+        /// The block size in bytes.
+        /// </summary>
+        [CodeGenMember("Size")]
+        public long SizeLong { get; }
+
+        /// <summary>
+        /// This property is a backwards-compatible facade for <see cref="SizeLong"/>,
+        /// which supports long values. Use <see cref="SizeLong"/> for full access of
+        /// supported values.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public int Size => checked((int)SizeLong);
```

Comment:
What does checked do?  Ensure we didn’t wrap around?

Reply:
Yeah. Without this, the extra bits are just discarded. Here, we actually throw the OverflowException.